packages/ecs/src/ComponentFunctions.ts

```diff
@@ -148,6 +148,14 @@ export interface ComponentPartial<
   onSet?: (entity: Entity, component: State<ComponentType>, json?: SetJSON) => void
   /** @todo Explain ComponentPartial.onRemove(...) */
   onRemove?: (entity: Entity, component: State<ComponentType>) => void | Promise<void>
+  /**
+   * If present, this function is called after this component is created by an extension in a GLTF node
+   *
+   * @param options GLTFParserOptions
+   * @param nodeIndex index of the node that has this component as an extension
+   * @returns
+   */
+  loadNode?: (options: any, nodeIndex: number) => Promise<void>
```

Comment:
we dont want to make this an official API in the ecs, since components can have anything assigned to it's definition. the way the gltf extensions work was kind of a hack regression back towards the original threejs gltf loader as the asynchronicity of reactors made it too hard to load things deterministically. at some point we will figure it out, but we should not be overloading the ecs with concepts from higher levels